scripts/livereduce_watchdog.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+echo "ARGS $*"
+
+# determine the configuration file.
+if [ $# -ge 1 ]; then
+    CONFIG_FILE="${1}"
+else
+    CONFIG_FILE=/etc/livereduce.conf
+fi
+
+# Check if the configuration file exists
+if [ ! -f "${CONFIG_FILE}" ]; then
+    echo "ERROR: Config file '${CONFIG_FILE}' not found." >&2
+    exit 1
+fi
+
+# --- CONFIG ---
+WATCHDOG_TARGET="/var/log/SNS_applications/livereduce.log"
+MANAGED_SERVICE="livereduce.service"
+# How often we check WATCHDOG_TARGET. Default is 60 seconds.
+INTERVAL="$(/bin/jq --raw-output '.watchdog.interval // 60' "${CONFIG_FILE}")"
+# Inactivity threshold. Default is 300 seconds.
+THRESHOLD="$(/bin/jq --raw-output '.watchdog.threshold // 300' "${CONFIG_FILE}")"
+
+WATCHDOG_LOG="/var/log/SNS_applications/livereduce_watchdog.log"
+# Track when we last issued a restart so we don't keep restarting every loop
+last_restart=0
+
+# Infinite loop
+while true; do
+
+  if [[ ! -e "$WATCHDOG_TARGET" ]]; then
```

Comment:
Should the watchdog service die in this case? Maybe it should be setup to (in the service file) add `After=livereduce.service` so livereduction is already running before this starts watching.